### PR TITLE
Add support for specifying multiple bind addresses

### PR DESCRIPTION
**Pull request by kyrias**

### CODE REVIEW

File: synapse/config/server.py [hunk 1]
```diff
@@ -155,9 +155,10 @@ def default_config(self, server_name, **kwargs):
             # The port to listen for HTTPS requests on.
             port: %(bind_port)s
 
-            # Local interface to listen on.
-            # The empty string will cause synapse to listen on all interfaces.
-            bind_address: ''
+            # Local addresses to listen on.
+            # This will listen on all IPv4 addresses by default.
```
Comment (ara4n) [inline synapse/config/server.py]: Can we have the example in here of how to listen on ipv6 too, even if commented out?  presumably

```
# - '::'
```

Comment (kyrias) [inline synapse/config/server.py]: I was thinking about that, but also need to specify that having both uncommented at the same time won't work on Linux, so I was thinking about whether to document this in the default config or somewhere else?

**Comment by ara4n** [comment]
this LGTM, modulo the minor doc change.  annoying that the binding setup is copypasted over all the different workers though, but not your fault...

File: synapse/config/server.py (hunk 1 above)
Comment (ara4n) [inline synapse/config/server.py]: it's going to be such a common scenario (plus we want to advertise that ipv6 is here, yay!) that i suspect it's worth spelling it out (assuming you are sure that the second bind will fail?)  
how about:

```
 # uncomment to listen on all IPv6 interfaces
 # N.B. on Linux this also listens on IPv4, so you'll need to comment out the line above.
 # - '::'
```

Comment (kyrias) [inline synapse/config/server.py]: On Linux you'll get an errno 98, yeah, annoyingly.

Comment added below.

**Comment by richvdh** [comment]
ok to test

**Comment by ara4n** [comment]
lgtm - going to merge and see what happens! thanks @kyrias!

**Pull request merged**